enhancements/sdk-migrate-bundle.md

```diff
@@ -0,0 +1,379 @@
+packagemanifests & bundles. This enhancement will add a new command, `migrate
+bundle` to the CLI. This new command will migrate an existing packagemanifest to
+a bundle.
+
+More information about the different format can be found at the URLs below:
+
+- Operator bundles: https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md
+- Package manifests format: https://github.com/operator-framework/operator-registry/tree/v1.5.3/#manifest-format
+
+The Operator SDK will traverse the given packagemanifest directory and for each
+of the versions create a new bundle.
+
+#### Generating bundle on disk
+
+In order to generate bundles to disk, we will refactor the `generate bundle`
+logic to be reusable in generating the bundles.
+
+===================== REWRITE BELOW =======================
+The `GenerateFunc` takes in the following arguments:
+
+- directory string - local directory where the manifests and metadata are
+  located
+- outputDir string - where manifests and metadata are copied
+- packageName string - the name of the package that bundle image belongs to
+- channels string - the list of channels that bundle image belongs to
+- channelDefault string - the default channel for the bundle image
+
+In order to use this function, we will obtain the required parameters from the
+listed *source*:
+
+| parameter | source |
+| --------- | ------ |
+| directory | packagemanifest directory |
+| outputDir | default `bundle` dir or flag `--output-dir` |
+| packageName | packageName from `*.package.yaml` |
+| channels | Any channel where `channels[].currentCSV == packagedCSV.Name` from `*.package.yaml` |
+| channelDefault | defaultChannel from `*.package.yaml` or `stable` |
+===================== REWRITE ABOVE =======================
+
+#### Creating bundle image
+
+If we choose to build bundle images, we will likely need a sane default build
+command but also offer an override.
+
+This would add 2 more flags to the CLI:
+
+- `[--image-base]` optional flag that indicates the base container name for
+  the bundles; e.g. `quay.io/example/memcached-operator-bundle`
+- `[--build-cmd]` optional flag that indicates the fully qualified build command;
+
+These flags would output a bundle image for each bundle created. We would use
+the `<dirname>` for the tags.
+
+The `--image-base` needs to be valid docker image name without the tag. We will
+use the bundle directory for the imagetag.
+
+Since we can't always anticipate what folks want to do with their builds, we'll
+have an excape hatch flag, `--build-cmd`, which will take in the full build
+command to run. This will include the builder ie. `podman`, `docker`, etc. this
+command will need to be in the `PATH` or they must supply the fully qualified
+path name.
+
+For example, `podman build -t quay.io/example/bundle ...`.
+
+#### Command Line Interface
+
+The new command line will look as follows:
+
+```
+operator-sdk pkgman-to-bundle <packagemanifestdir> [--build-image=] \
```

Comment:
can we find a better name than `pkgman-to-bundle` ? Ideas? Also, could not be a subcommand of `operator-sdk bundle`? E.g `operator-sdk bundle migration `

Reply:
We went back and forth on this name. The reason it is `pkgman-to-bundle` it was descriptive and we know it will be removed in 2.0. The other ideas were `migrate packagemanifest`, `migrate bundle`, `convert packagemanifests`, we didn't want to waste `migrate` nor `convert` with only one subcommand that will go away soon.